src/devices/Vl53L1X/Vl53L1X.cs

```diff
@@ -0,0 +1,959 @@
+                return
+                    (WindowDetectionMode)(temp &
+                                          0x7);
+            }
+        }
+
+        /// <summary>
+        /// Returns the low threshold in mm.
+        /// </summary>
+        public ushort DistanceThresholdLow
+        {
+            get
+            {
+                return ReadUInt16((ushort)Registers.SYSTEM__THRESH_LOW);
+            }
+        }
+
+        /// <summary>
+        /// Returns the high threshold in mm.
+        /// </summary>
+        public ushort DistanceThresholdHigh
+        {
+            get
+            {
+                return ReadUInt16((ushort)Registers.SYSTEM__THRESH_HIGH);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the <see cref="Roi" />, the position of which is centered about the optical center.
+        /// The smallest acceptable ROI size is 4.
+        /// </summary>
+        public Roi Roi
+        {
+            get
+            {
+                byte temp = ReadByte((ushort)Registers.ROI_CONFIG__USER_ROI_REQUESTED_GLOBAL_XY_SIZE);
+
+                return new Roi((ushort)((temp & 0x0F) + 1), (ushort)(((temp & 0xF0) >> 4) + 1));
+            }
+            set
+            {
+                byte opticalCenter = ReadByte((ushort)Registers.VL53L1X_ROI_CONFIG__MODE_ROI_CENTRE_SPAD);
+                ushort x = value.Width;
+                ushort y = value.Height;
+
+                if (x > 16)
+                {
+                    x = 16;
+                }
+
+                if (y > 16)
+                {
+                    y = 16;
+                }
+
+                if (x > 10 || y > 10)
+                {
+                    opticalCenter = 199;
+                }
+
+                WriteRegister((ushort)Registers.ROI_CONFIG__USER_ROI_CENTRE_SPAD, opticalCenter);
+                WriteRegister((ushort)Registers.ROI_CONFIG__USER_ROI_REQUESTED_GLOBAL_XY_SIZE,
+                    (byte)(((y - 1) << 4) | (x - 1)));
+            }
+        }
+
+        /// <summary>
+        /// This function programs the new user ROI center, please to be aware that there is no check in this function
+        /// if the ROI center vs ROI size is out of border.
+        /// </summary>
+        public byte RoiCenter
+        {
+            get
+            {
+                return ReadByte((ushort)Registers.ROI_CONFIG__USER_ROI_CENTRE_SPAD);
+            }
+            set
+            {
+                WriteRegister((ushort)Registers.ROI_CONFIG__USER_ROI_CENTRE_SPAD, value);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a new signal threshold in kcps where the default is 1024 kcps.
+        /// </summary>
+        public ushort SignalThreshold
+        {
+            get
+            {
+                return (ushort)(ReadUInt16((ushort)Registers.RANGE_CONFIG__MIN_COUNT_RATE_RTN_LIMIT_MCPS) << 3);
+            }
+            set
```

Comment:
nit, you need an extra CR between } and set.

Reply:
Done